tests/PHPStan/Command/ErrorFormatter/JsonErrorFormatterTest.php

```diff
@@ -2,118 +2,241 @@
-		$style = new ErrorsConsoleStyle(new StringInput(''), $outputStream);
-
-		$this->assertSame(1, $formatter->formatErrors($analysisResult, $style));
-
-		rewind($outputStream->getStream());
-		$output = stream_get_contents($outputStream->getStream());
+		yield [
+			'One generic error',
+			1,
+			0,
+			1,
+			'
+{
+	"totals":{
+		"errors":1,
+		"file_errors":0
+	},
+	"files":[],
+	"errors": [
+		"first generic error"
+	]
+}',
+		];
 
-		$expected = '
+		yield [
+			'Multiple file errors',
+			1,
+			4,
+			0,
+			'
 {
 	"totals":{
 		"errors":0,
-		"file_errors":2
+		"file_errors":4
 	},
 	"files":{
-		"foo.php":{
-			"errors":1,
+		"/data/folder/with space/and unicode 😃/project/folder with unicode 😃/file name with \"spaces\" and unicode 😃.php":{
+			"errors":2,
 			"messages":[
 				{
-					"message":"Foo",
-					"line":1,
-					"ignorable":true
+					"message": "Bar",
+					"line": 2,
+					"ignorable": true
+				},
+				{
+					"message": "Foo",
+					"line": 4,
+					"ignorable": true
 				}
 			]
 		},
-		"file name with \"spaces\" and unicode 😃.php":{
-			"errors":1,
+		"/data/folder/with space/and unicode 😃/project/foo.php":{
+			"errors":2,
 			"messages":[
 				{
-					"message":"Bar",
-					"line":2,
-					"ignorable":true
+					"message": "Foo",
+					"line": 1,
+					"ignorable": true
+				},
+				{
+					"message": "Bar",
+					"line": 5,
+					"ignorable": true
 				}
 			]
 		}
 	},
 	"errors": []
-}
-';
-		$this->assertJsonStringEqualsJsonString($expected, $output);
+}',
+		];
+
+		yield [
+			'Multiple generic errors',
+			1,
+			0,
+			2,
+			'
+{
+	"totals":{
+		"errors":2,
+		"file_errors":0
+	},
+	"files":[],
+	"errors": [
+		"first generic error",
+		"second generic error"
+	]
+}',
+		];
+
+		yield [
+			'Multiple file, multiple generic errors',
+			1,
+			4,
+			2,
+			'
+{
+	"totals":{
+		"errors":2,
+		"file_errors":4
+	},
+	"files":{
+		"/data/folder/with space/and unicode 😃/project/folder with unicode 😃/file name with \"spaces\" and unicode 😃.php":{
+			"errors":2,
+			"messages":[
+				{
+					"message": "Bar",
+					"line": 2,
+					"ignorable": true
+				},
+				{
+					"message": "Foo",
+					"line": 4,
+					"ignorable": true
+				}
+			]
+		},
+		"/data/folder/with space/and unicode 😃/project/foo.php":{
+			"errors":2,
+			"messages":[
+				{
+					"message": "Foo",
+					"line": 1,
+					"ignorable": true
+				},
+				{
+					"message": "Bar",
+					"line": 5,
+					"ignorable": true
+				}
+			]
+		}
+	},
+	"errors": [
+		"first generic error",
+		"second generic error"
+	]
+}',
+		];
 	}
 
 	/**
-	 * @dataProvider dataPretty
-	 * @param bool $pretty
+	 * @param string $message          Test message
+	 * @param int    $exitCode         Expected exit code from the application
+	 * @param int    $numFileErrors    Number of errors for file
+	 * @param int    $numGenericErrors Number of generic errors
+	 * @param string $expected         Expected output
+	 *
+	 * @dataProvider jsonOutputProvider
```

Comment:
Hi, please respect the coding standards of the rest of the codebase. Namely:

* No parameter descriptions (and especially not intended like that!)
* Data provider always starts with `data`
* `@dataProvider` is always first.

Reply:
My problem is that those descriptions are useless, see:

```
@param int    $numGenericErrors Number of generic errors
```

Reply:
@ondrejmirtes fixed in https://github.com/phpstan/phpstan/pull/1034/commits/515d21ba378154041c4a9f7e28e854f2b02e57aa